src/dmd/ctfeexpr.d

```diff
@@ -1408,6 +1408,34 @@ bool ctfeIdentity(const ref Loc loc, EXP op, Expression e1, Expression e2)
         complex_t v2 = e2.toComplex();
         cmp = RealIdentical(creall(v1), creall(v2)) && RealIdentical(cimagl(v1), cimagl(v1));
     }
+    else if (e1.op == EXP.structLiteral || e2.op == EXP.structLiteral)
```

Comment:
What's all this struct literal business, does it have anything to do with floating-point identity?

Reply:
Not sure, I did not originally write it, I rebased Iain's PR which rebased another PR from 2013. I removed it and modified `ctfeRawCmp` now.